src/Prefix.jl

```diff
@@ -597,20 +597,25 @@ function get_addable_spec(name::AbstractString, version::VersionNumber;
         uuid=uuid,
         #version=version,
         tree_hash=tree_hash_sha1,
-        repo=Pkg.Types.GitRepo(rev=git_commit_sha, source=valid_url),
+        rev=git_commit_sha,
+        url=valid_url,
```

Comment:
Specifying `repo` directly isn't compatible with the public `Pkg.add` API as `Pkg.handle_package_input!` doesn't expect it to exist, instead constructs it from `rev` and `url` etc.

Reply:
Improved handling https://github.com/JuliaLang/Pkg.jl/pull/4170